cmd/quotas.go

```diff
@@ -130,11 +133,81 @@ var quotasCmd = &cobra.Command{
 	},
 }
 
+var defaultResources = corev1.ResourceList{
+	// CPU
+	"requests.cpu": *resource.NewQuantity(70, resource.DecimalSI),
+	"limits.cpu":   *resource.NewQuantity(70, resource.DecimalSI),
+
+	// Memory
+	"requests.memory": *resource.NewScaledQuantity(368, resource.Giga),
+	"limits.memory":   *resource.NewScaledQuantity(368, resource.Giga),
+
+	// Storage
+	"requests.storage": *resource.NewScaledQuantity(4, resource.Tera),
+
+	// GPU
+	"requests.nvidia.com/gpu": *resource.NewQuantity(2, resource.DecimalSI),
+
+	// Pods
+	"pods": *resource.NewQuantity(100, resource.DecimalSI),
+
+	// Services
+	"services.nodeports":     *resource.NewQuantity(0, resource.DecimalSI),
+	"services.loadbalancers": *resource.NewQuantity(0, resource.DecimalSI),
+}
+
+func overrideResourceQuotas(profile *kubeflowv1.Profile) {
+
+	// if we iterate over defaultResources, won't be able to find the key since default key is requests.storage and profileLabels key: quotas.statcan.gc.ca/requests.storage
+	// used strings.Contains()
+	// Also defaultResources has different resource dataTypes (for ex. resource.Tera, resource.Giga etc.) ; need checks on which resource quota you're overriding
+
+	for key, _ := range defaultResources { // or can iterate over profile.labels to avoid hard code of profiles label
+		// CPU
+		if strings.Contains(profile.Labels["quotas.statcan.gc.ca/requests.cpu"], key.String()) {
+			cpuRequest, _ := strconv.Atoi(profile.Labels["quotas.statcan.gc.ca/requests.cpu"])
+			defaultResources[key] = *resource.NewQuantity(int64(cpuRequest), resource.DecimalSI)
+		}
+		if strings.Contains(profile.Labels["quotas.statcan.gc.ca/limits.cpu"], key.String()) {
+			cpuLimit, _ := strconv.Atoi(profile.Labels["quotas.statcan.gc.ca/limits.cpu"])
+			defaultResources[key] = *resource.NewQuantity(int64(cpuLimit), resource.DecimalSI)
+		}
+		// Memory
+		if strings.Contains(profile.Labels["quotas.statcan.gc.ca/requests.memory"], key.String()) {
+			memoryRequest, _ := strconv.Atoi(profile.Labels["quotas.statcan.gc.ca/requests.memory"])
+			defaultResources[key] = *resource.NewScaledQuantity(int64(memoryRequest), resource.Giga)
+		}
+		if strings.Contains(profile.Labels["quotas.statcan.gc.ca/limits.memory"], key.String()) {
+			memoryLimit, _ := strconv.Atoi(profile.Labels["quotas.statcan.gc.ca/limits.memory"])
+			defaultResources[key] = *resource.NewScaledQuantity(int64(memoryLimit), resource.Giga)
```

Comment:
I think there is probably a parser for kubernetes resources that would let you write "2Gi" and parse that into the correct datatype. It's got to be used in the kubernetes library.

Reply:
There will work https://pkg.go.dev/k8s.io/apimachinery/pkg/api/resource#MustParse.

Reply:
It will be better to use NewScaledQuantity and NewQuantity. These functions use a format parameter. The output looks like:`5G` . Using value * 10^scale in DecimalSI format https://pkg.go.dev/k8s.io/apimachinery/pkg/api/resource#NewScaledQuantity
Whereas, MustParse outputs the value in this format: `5000000000 (DecimalSI)`